crates/typst-ide/src/jump.rs

```diff
@@ -115,7 +115,7 @@ pub fn jump_from_cursor(
     source: &Source,
     cursor: usize,
 ) -> Option<Position> {
-    let node = LinkedNode::new(source.root()).leaf_at(cursor)?;
+    let node = LinkedNode::new(source.root()).leaf_at(cursor, Side::After)?;
```

Comment:
I'm not sure which side is correct here tbh. If we have a text node, I think jumping from either side should work ideally. Perhaps we should try after first and if it doesn't yield a text node, then try before?

Reply:
I don't really know either... after should preserve the current behaviour, but I'm not sure if that's correct. Is this used by CodeMirror in the webapp? It doesn't look like typst-lsp uses it, so we don't have to consider any immediate impacts there.

Reply:
Also if we decide that after then before is the right approach, we could do that like this:

```suggestion
    let root = LinkedNode::new(source.root());
    let node = root.leaf_at(cursor, Side::After).or_else(|| root.leaf_at(cursor, Side::Before))?;
```

Reply:
It is used by the web app when clicking the little arrow to the top right of the editor, and also automatically all the time if scroll-on-type is enabled. 

> after should preserve the current behaviour

Isn't `Before` the old behaviour?

> Also if we decide that after then before is the right approach, we could do that like this:

This looks good.

Reply:
No wait, this isn't enough. We should only keep the `After` node if it is indeed a Text node. But we can also keep the old behaviour and I'll look into checking both sides separately.

Reply:
> Isn't `Before` the old behaviour?

Yeah you're right! I've switched it to use that in 6a43276a739a11d08329b36e2ec8f12bce368c16.

> No wait, this isn't enough. We should only keep the `After` node if it is indeed a Text node. But we can also keep the old behaviour and I'll look into checking both sides separately.

Oh true! Here's one potential approach, feel free to wait till later if you'd prefer though:

```rust
    fn is_text(node: &LinkedNode) -> bool {
        node.get().kind() == SyntaxKind::Text
    }

    let root = LinkedNode::new(source.root());
    let node = root
        .leaf_at(cursor, Side::Before)
        .filter(is_text)
        .or_else(|| root.leaf_at(cursor, Side::After))
        .filter(is_text)?;
```

Reply:
This looks good, you can add it here.

Reply:
Ok, added in 74923b1f54742dc67dd4bafab28620de4621ca60.